pkg/manager/internal/controllers/organizationroletemplate_controller.go

```diff
@@ -62,6 +98,10 @@ func (r *OrganizationRoleTemplateReconciler) Reconcile(req ctrl.Request) (ctrl.R
 			Reason:  "SetupComplete",
 			Message: "OrganizationRoleTemplate setup is complete.",
 		})
+		isChanged = true
+	}
+
+	if isChanged {
 		if err := r.Status().Update(ctx, organizationRoleTemplate); err != nil {
```

Comment:
isn't this a no-op on the API server if things haven't changed? Sure we're issuing API call, but it isn't so bad compared to all other API calls we're performing during this execution.

Reply:
I don't want to issue API call every time, and only issue that when something is changed, because:
1. We had a discussion in when I built KubeCarrier IIRC, and we don't want to issue API calls everytime.
2. If we always issue API calls, it will cause so many request failures with resource version conflicts, and leads to many retries, which I don't want to.

Reply:
I don't really know :) 
I think it also depends on how many organizations we have in the system :)